packages/e2e-tests/config/setup-playwright.js

```diff
@@ -0,0 +1,183 @@
+		) {
+			return;
+		}
+
+		// Viewing posts on the front end can result in this error, which
+		// has nothing to do with Gutenberg.
+		if ( text.includes( 'net::ERR_UNKNOWN_URL_SCHEME' ) ) {
+			return;
+		}
+
+		// Network errors are ignored only if we are intentionally testing
+		// offline mode.
+		if (
+			text.includes( 'net::ERR_INTERNET_DISCONNECTED' ) &&
+			isOfflineMode()
+		) {
+			return;
+		}
+
+		// As of WordPress 5.3.2 in Chrome 79, navigating to the block editor
+		// (Posts > Add New) will display a console warning about
+		// non - unique IDs.
+		// See: https://core.trac.wordpress.org/ticket/23165
+		if ( text.includes( 'elements with non-unique id #_wpnonce' ) ) {
+			return;
+		}
+
+		// As of WordPress 5.3.2 in Chrome 79, navigating to the block editor
+		// (Posts > Add New) will display a console warning about
+		// non - unique IDs.
+		// See: https://core.trac.wordpress.org/ticket/23165
+		if ( text.includes( 'elements with non-unique id #_wpnonce' ) ) {
+			return;
+		}
+
+		const logFunction = OBSERVED_CONSOLE_MESSAGE_TYPES[ type ];
+
+		// As of Puppeteer 1.6.1, `message.text()` wrongly returns an object of
+		// type JSHandle for error logging, instead of the expected string.
+		//
+		// See: https://github.com/GoogleChrome/puppeteer/issues/3397
+		//
+		// The recommendation there to asynchronously resolve the error value
+		// upon a console event may be prone to a race condition with the test
+		// completion, leaving a possibility of an error not being surfaced
+		// correctly. Instead, the logic here synchronously inspects the
+		// internal object shape of the JSHandle to find the error text. If it
+		// cannot be found, the default text value is used instead.
+		text = get(
+			message.args(),
+			[ 0, '_remoteObject', 'description' ],
+			text
+		);
+
+		// Disable reason: We intentionally bubble up the console message
+		// which, unless the test explicitly anticipates the logging via
+		// @wordpress/jest-console matchers, will cause the intended test
+		// failure.
+
+		// eslint-disable-next-line no-console
+		console[ logFunction ]( text );
+	} );
+}
+
+/**
+ * Navigates to the post listing screen and bulk-trashes any posts which exist.
+ *
+ * @param {string} postType - String slug for type of post to trash.
+ *
+ * @return {Promise} Promise resolving once posts have been trashed.
+ */
+export async function trashExistingPosts( postType = 'post' ) {
+	await switchUserToAdmin();
+	// Visit `/wp-admin/edit.php` so we can see a list of posts and delete them.
+	const query = addQueryArgs( '', {
+		post_type: postType,
+	} ).slice( 1 );
+	await visitAdminPage( 'edit.php', query );
+
+	// If this selector doesn't exist there are no posts for us to delete.
+	const bulkSelector = await page.$( '#bulk-action-selector-top' );
+	if ( ! bulkSelector ) {
+		return;
+	}
+
+	// Select all posts.
+	await page.waitForSelector( '[id^=cb-select-all-]' );
```

Comment:
Small nit: but you can actually take out this line to capitalize [on all that Playwright auto-waiting goodness](https://playwright.dev/docs/actionability) :)

Reply:
Removed in https://github.com/WordPress/gutenberg/pull/34089/commits/0af07566c88de19649a8f081cb2b1a92e34423ab

Reply:
I don't think you can though, it's not always the same, and removing it might introduce flaky results. The auto-waiting mechanism Playwright supports does wait for several actionability checks, but it seems like it won't retry it. Taking below as an example.

```js
// 1. Start from a blank page.
await page.setContent('');

// 2. Click on a button that not yet exist.
await page.click('button');

// 3. Attach the button to the DOM.
await page.setContent('<button>Button</button>');
```

Step 3 will never be reached as Playwright will just wait for step 2 to finish but it won't.

I think this is an important gotcha to note when migrating to Playwright, better to not have false interpretation of the benefits they offer.

Reply:
This example is correct - it will never reach step 3. I don't think it's a correct representation of the potential issue though, as we never create DOM nodes from within the test script itself, but asynchronously from the application that's running underneath. So, in order to provide a real-world example of a situation where a node is attached **after** the click waiter has been dispatched, we need to do something like this:

```js
// 1. Start from a blank page.
await page.setContent('');

// 2. Attach the button to the DOM asynchronously (doesn't stop test execution).
setTimeout( () => {
	page.setContent( '<button>Howdy!</button>' );
}, 2000 );

// 3. Click/start waiting on a button that doesn't exist yet.
await page.click('button');
```

If you run this example in a debugging mode, you'll see that the `click` is dispatched first, then the button is attached, and in the end, the button is actually clicked. Does that make sense?

Reply:
Oops, Yes! My example is wrong 😅. Haven't tested the example you provided, but I trust you 👍 .